src/oneD/StFlow.cpp

```diff
@@ -517,14 +481,99 @@ void StFlow::evalResidual(double* x, double* rsd, int* diag,
                 rsd[index(c_offset_T, j)] = T(x,j) - T_fixed(j);
                 diag[index(c_offset_T, j)] = 0;
             }
+        }
+    }
+}
 
-            if (m_usesLambda) {
-                rsd[index(c_offset_L, j)] = lambda(x, j) - lambda(x, j - 1);
-            } else {
-                rsd[index(c_offset_L, j)] = lambda(x, j);
+void StFlow::evalSpecies(double* x, double* rsd, int* diag,
+                         double rdt, size_t jmin, size_t jmax)
+{
+    for (size_t j = jmin; j <= jmax; j++) {
```

Comment:
similarly, pull logic out, so you get logic -> loop over interior points -> loop over species.